internal/provider/kubernetes/controller.go

```diff
@@ -833,7 +833,7 @@ func (r *gatewayAPIReconciler) processBackendTrafficPolicies(ctx context.Context
 		policy := policy
 		// Discard Status to reduce memory consumption in watchable
 		// It will be recomputed by the gateway-api layer
-		policy.Status = egv1a1.BackendTrafficPolicyStatus{}
+		policy.Status = gwapiv1a2.PolicyStatus{}
```

Comment:
should we modify this like #2845?

Reply:
I believe so, the status can't be discarded because other controllers may also modified it.

Reply:
got it!